DynamicTablesPkg/Library/Common/TableHelperLib/ConfigurationManagerObjectParser.c

```diff
@@ -871,82 +871,51 @@ PrintStringPtr (
     String = (UINT8 *)"(NULLPTR)";
   }
 
-  PrintString (Format, String);
+  PrintString (Format, String, Length);
 }
 
-/** Print 4 characters.
+/** Print characters.
 
   @param [in]  Format  Format to print the Ptr.
   @param [in]  Ptr     Pointer to the characters.
+  @param [in]  Length  Length of the field
 **/
 STATIC
 VOID
 EFIAPI
-PrintChar4 (
+PrintChars (
   IN  CONST CHAR8  *Format,
-  IN  UINT8        *Ptr
+  IN  UINT8        *Ptr,
+  IN  UINT32       Length
   )
 {
-  DEBUG ((
-    DEBUG_ERROR,
-    (Format != NULL) ? Format : "%c%c%c%c",
-    Ptr[0],
-    Ptr[1],
-    Ptr[2],
-    Ptr[3]
-    ));
-}
-
-/** Print 6 characters.
+  UINT32  Index;
 
-  @param [in]  Format  Format to print the Ptr.
-  @param [in]  Ptr     Pointer to the characters.
-**/
-STATIC
-VOID
-EFIAPI
-PrintChar6 (
-  IN  CONST CHAR8  *Format,
-  IN  UINT8        *Ptr
-  )
-{
-  DEBUG ((
-    DEBUG_ERROR,
-    (Format != NULL) ? Format : "%c%c%c%c%c%c",
-    Ptr[0],
-    Ptr[1],
-    Ptr[2],
-    Ptr[3],
-    Ptr[4],
-    Ptr[5]
-    ));
+  for (Index = 0; Index < Length; Index++) {
+    DEBUG ((DEBUG_INFO, "%c", Ptr[Index]));
+  }
 }
 
-/** Print 8 characters.
+/** Dump data in Hex format
 
   @param [in]  Format  Format to print the Ptr.
-  @param [in]  Ptr     Pointer to the characters.
+  @param [in]  Ptr     Pointer to the string.
+  @param [in]  Length  Length of the field
 **/
 STATIC
 VOID
 EFIAPI
-PrintChar8 (
-  IN  CONST CHAR8  *Format,
-  IN  UINT8        *Ptr
+HexDump (
+  IN CONST CHAR8  *Format,
+  IN UINT8        *Ptr,
+  IN UINT32       Length
   )
 {
-  DEBUG ((
-    DEBUG_ERROR,
-    (Format != NULL) ? Format : "%c%c%c%c%c%c%c%c",
-    Ptr[0],
-    Ptr[1],
-    Ptr[2],
-    Ptr[3],
-    Ptr[4],
-    Ptr[5],
-    Ptr[6],
-    Ptr[7]
-    ));
+  UINT32  Index;
+
+  for (Index = 0; Index < Length; Index++) {

```

Comment:
Shouldn't we set:
```
UINT32 Count;

Count = Length /  sizeof (UINT64);
for (...; Index < Count; ...)
```

and maybe rename the function PrintHexUint64 () as we are expecting to receive 64 bits values

Reply:
Hi Pierre,
Thanks for reviewing. 

Actually, I meant to add this HexDump function to dump arbitrary length array.  Later, I found this patch had dependency on other patches. Hence, I have made another PR: https://github.com/tianocore/edk2/pull/5827. Could you please review 5827 instead?

Sorry, I am new to this and will find a way to abandon this PR.